### DISCUSSION TITLE
Lilygo T-Display-S3

### DISCUSSION THREAD

**Discussion by scruss** [micropython/micropython#9329] [Hardware & Peripherals] [answered]
I've had limited success flashing the Lilygo [T-Display-S3](https://www.lilygo.cc/products/t-display-s3). Most of the time, the board isn't recognized as a USB device under Linux:
```
[612761.947071] usb usb2-port2: attempt power cycle
[612762.811001] usb 2-2: new full-speed USB device number 51 using xhci_hcd
[612762.811025] usb 2-2: Device not responding to setup address.
[612763.019036] usb 2-2: Device not responding to setup address.
[612763.227012] usb 2-2: device not accepting address 51, error -71
[612763.563046] usb 2-2: new full-speed USB device number 52 using xhci_hcd
[612763.582940] usb 2-2: Device not responding to setup address.
[612763.791037] usb 2-2: Device not responding to setup address.
[612763.999001] usb 2-2: device not accepting address 52, error -71
[612763.999066] usb usb2-port2: unable to enumerate USB device
[612764.451024] usb 2-2: new full-speed USB device number 53 using xhci_hcd
```

**Comment by russhughes** [comment]
In my case the issue was the PSRAM configuration was wrong (it’s octal) and the device crashes disabling the USB port. Once I changed the PSRAM type to octal every started working.

**Reply by scruss** [reply]
Where did you configure this value? Did you use the generic S3 configuration?

Also, I can't seem to get the board out of the crashed USB state (tried holding/releasing various combinations of BOOT and RESET) to reprogram it

**Reply by russhughes** [reply]
I used the GENERIC_S3_SPIRAM folder to create a new T-DISPLAY-S3 board definition and removed the reference to boards/sdkconfig.spiram_sx in the T-DISPLAY-S3/mpconfigboard.cmake file. I added the contents of the boards/sdkconfig.spiram_sx to the T-DISPLAY-S3/sdkconfig.board file and changed the settings to match the board.

```
CONFIG_FLASHMODE_QIO=y
CONFIG_ESPTOOLPY_FLASHFREQ_80M=y
CONFIG_ESPTOOLPY_FLASHSIZE_DETECT=y
CONFIG_ESPTOOLPY_AFTER_NORESET=y

CONFIG_SPIRAM_MEMTEST=

CONFIG_ESPTOOLPY_FLASHSIZE_4MB=
CONFIG_ESPTOOLPY_FLASHSIZE_8MB=
CONFIG_ESPTOOLPY_FLASHSIZE_16MB=y
CONFIG_PARTITION_TABLE_CUSTOM=y
CONFIG_PARTITION_TABLE_CUSTOM_FILENAME="partitions-16MiB.csv"

CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG=y

CONFIG_ESP32S3_SPIRAM_SUPPORT=y
CONFIG_SPIRAM_MODE_OCT=y
CONFIG_SPIRAM_TYPE_AUTO=y
CONFIG_DEFAULT_PSRAM_CLK_IO=30
CONFIG_DEFAULT_PSRAM_CS_IO=26
CONFIG_SPIRAM_SPEED_80M=y
CONFIG_SPIRAM=y
CONFIG_SPIRAM_BOOT_INIT=y
CONFIG_SPIRAM_IGNORE_NOTFOUND=y
CONFIG_SPIRAM_USE_MEMMAP=y
CONFIG_SPIRAM_MEMTEST=y
CONFIG_SPIRAM_MALLOC_ALWAYSINTERNAL=16384
CONFIG_SPIRAM_MALLOC_RESERVE_INTERNAL=32768

```

I have not had much time to test, but it connects now and allows me use the PSRAM. I will probably create a boards/sdkconfig.spiram_oct and use it instead of dumping the config into sdkconfig.board.

**Reply by russhughes** [reply]
To reprogram, press and hold the boot button, then press and release reset, and finally release the boot button.

**Comment by scruss** [comment, marked as answer]
Unless you're willing to write a custom PSRAM configuration, Lilygo recommends the GENERIC_S3 binary, which after many attempts to reset and reflash, I've finally managed to get running:

```
MicroPython v1.19.1 on 2022-10-11; ESP32S3 module with ESP32S3
Type "help()" for more information.
>>> 
```

These are the commands I used to clear flash and reprogram it:

```
esptool.py --chip esp32s3 --port /dev/ttyACM0 erase_flash
esptool.py --chip esp32s3 --port /dev/ttyACM0 write_flash -z 0  GENERIC_S3-20221011-unstable-v1.19.1-543-gab317a0d6.bin
```
